dace/codegen/targets/fpga.py

```diff
@@ -2002,6 +2002,13 @@ def _generate_MapEntry(self, sdfg, dfg, state_id, node, function_stream, callsit
                         # This could occur, for example, if they are defined in terms of other symbols, which
                         # is the case in a tiled map
                         pass
+                    
+                    # To enforce opencl type long instead of c type long long for intel fpga
+                    v = dace.config.Config.get("compiler", "fpga", "vendor")
+                    if v.casefold() == 'intel_fpga'.casefold():
+                        if loop_var_type.__contains__("long long"):
```

Comment:
Better to use `substring in string` for the check.
You can also apply the replace directly, it will not change anything if it does not found the substring